### PR TITLE
Added N-queens.py

### PR DESCRIPTION
**Pull request by Pranav016**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added N-queens problem in python.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1475

### CODE REVIEW

**Comment by charlie219** [comment]
@Pranav016 update the `README.md` file by making a new heading `Backtracking` and add the link to your code there

**Comment by Pranav016** [comment]
> @Pranav016 update the `README.md` file by making a new heading `Backtracking` and add the link to your code there

Oh I forgot, I'll do it rn.

File: Python/backtracking/N-queens.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# You are given N, and for a given N x N chessboard, find a way to place N queens such that no queen can attack any other queen on the chess board. A queen can be killed when it lies in the same row, or same column, or the same diagonal of any of the other queens. You have to print all such configurations.
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py]: Gosh! That's a very very long comment! 

We usually assume that the Code should be formatted enough to be readable enough. You can try breaking down this Comment into multiple lines.

File: Python/backtracking/N-queens.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Sample input
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py]: Share sample Input and Output at the end of the Code.

File: Python/backtracking/N-queens.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        if isSafeToPut(i,j,n,solution):
+            if nqueen(solution,n,i+1):
+                return True
+        solution[i][j]=0
+
+    return False
+    
+
+
+# main
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py]: Enclose the main in a proper `if __name__ == '__main__'`

File: Python/backtracking/N-queens.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    for j in range(n):
+        solution[i][j]=1
+        if isSafeToPut(i,j,n,solution):
+            if nqueen(solution,n,i+1):
+                return True
+        solution[i][j]=0
+
+    return False
+    
+
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py:53]: Remove extra lines from the Code

File: Python/backtracking/N-queens.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+def nqueen(solution,n,i):
+    if i==n:
+        for i in range(n):
+            for j in range(n):
+                print(solution[i][j],end=" ")
+        print()
+        return False
+
+    for j in range(n):
+        solution[i][j]=1
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py:47]: Can you try formatting your Code? If you are using an IDE like PyCharm, you can do that automatically via `Ctrl+Alt+L`

File: Python/backtracking/N-queens.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
+# You are given N, and for a given N x N chessboard, find a way to place N queens such that no queen can attack any other queen on the chess board. A queen can be killed when it lies in the same row, or same column, or the same diagonal of any of the other queens. You have to print all such configurations.
+
+# Sample input
+# 4
+
+def isSafeToPut(i,j,n,solution):
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py:3]: Add proper Docstrings for functions to explain their functionality

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by charlie219** [comment]
@HarshCasper I was assigned similar issue for #1524 
Please have a look at my PR and consider the best solution
PR #1538

**Comment by Pranav016** [comment]
@HarshCasper  Sure, I'll make all the suggested changes. Thanks for the feedback , it'll help me improve😄

**Comment by Pranav016** [comment]
> @HarshCasper I was assigned similar issue for #1524
> Please have a look at my PR and consider the best solution
> PR #1538

Hello @charlie219 , please refer to this issue #1475 that was raised 4 days ago. Do follow the [CODE_OF_CONDUCT](https://github.com/TesseractCoding/NeoAlgo/blob/master/CODE_OF_CONDUCT.md) and check existing issues before any PRs and respect others time :)

**Comment by charlie219** [comment]
@Pranav016 there is already a section for backtracking in `REDME.md`, I think it's useless to create a separate heading or it.
Please update this.

**Comment by Pranav016** [comment]
> @Pranav016 there is already a section for backtracking in `REDME.md`, I think it's useless to create a separate heading or it.
> Please update this.

That heading got merged to master after this PR was made. I rebased then I noticed. I'll update it..

**Comment by Pranav016** [comment]
> > @Pranav016 there is already a section for backtracking in `REDME.md`, I think it's useless to create a separate heading or it.
> > Please update this.
> 
> That heading got merged to master after this PR was made. I rebased then I noticed. I'll update it..

Updated!

File: Python/backtracking/N-queens.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# Problem statement- You are given N, and for a given N x N chessboard. We have to place all queens such that they are not in the same row, or same column, or the same diagonal of any of the other queens.
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py:1]: Modularize this. It is not readable enough. Break down the same using Multiline Strings

File: Python/backtracking/N-queens.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    n=int(input())
+    solution=[[0 for j in range(n)]for i in range(n)]
+    nqueen(solution,n,0)
+
+# Sample input-
+# 4
+
+# Sample output-
+# 0 1 0 0 0 0 0 1 1 0 0 0 0 0 1 0 
+# 0 0 1 0 1 0 0 0 0 0 0 1 0 1 0 0
```
Comment (HarshCasper) [inline Python/backtracking/N-queens.py:65]: Use Multi-Line String instead of Single Line Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny changes. Would be good to merge after this 💯

**Comment by charlie219** [comment]
@HarshCasper, there already exist N-Queens under Backtracking in Python. I don't this its right to add another similar code.

**Comment by Pranav016** [comment]
> @HarshCasper, there already exist N-Queens under Backtracking in Python. I don't this its right to add another similar code.

@charlie219 I had raised an issue for this problem 10 days ago, refer to #1475 .
I just checked an issue where you commented 8 days ago(same day this PR was made) that you want to work on the same problem, refer to #1524 

Please check existing issues and pull requests before you make a PR, to make sure Contribution guidelines are being followed and value other's time as well.

**Comment by Pranav016** [comment]
> > @HarshCasper, there already exist N-Queens under Backtracking in Python. I don't this its right to add another similar code.
> 
> @charlie219 I had raised an issue for this problem 10 days ago, refer to #1475 .
> I just checked an issue where you commented 8 days ago(same day this PR was made) that you want to work on the same problem, refer to #1524
> 
> Please check existing issues and pull requests before you make a PR, to make sure Contribution guidelines are being followed and value other's time as well.

@charlie219

**Pull request closed without merging**